src/synthesize.jl

```diff
@@ -85,39 +81,42 @@ function synthesize(atm::ModelAtmosphere, linelist, λ_start, λ_stop, λ_step=0
     else
         StepRangeLen(λ_start, λ_step, Int(round((λ_stop - λ_start)/λ_step))+1)
     end
-    synthesize(atm, linelist, wls; kwargs...)
+    synthesize(atm, linelist, A_X, wls; kwargs...)
 end
-function synthesize(atm::ModelAtmosphere, linelist, λs::AbstractRange; metallicity::Real=0.0, 
-                    vmic::Real=1.0, abundances::Dict{String, <:Real}=Dict{String, Float64}(), 
-                    line_buffer::Real=10.0, cntm_step::Real=1.0, hydrogen_lines=true, 
-                    mu_grid=0:0.05:1, line_cutoff_threshold=1e-3,
-                    solar_abundances=asplund_2020_solar_abundances, solar_relative=true,
+function synthesize(atm::ModelAtmosphere, linelist, A_X::Vector{<:Real}, λs::AbstractRange; 
+                    vmic::Real=1.0, line_buffer::Real=10.0, cntm_step::Real=1.0, 
+                    hydrogen_lines=true, mu_grid=0:0.05:1, line_cutoff_threshold=1e-3,
                     ionization_energies=ionization_energies, 
                     partition_funcs=partition_funcs, equilibrium_constants=equilibrium_constants)
     #work in cm
     λs = λs * 1e-8
     cntm_step *= 1e-8
     line_buffer *= 1e-8
     cntmλs = (λs[1] - line_buffer - cntm_step) : cntm_step : (λs[end] + line_buffer + cntm_step)
+    sorted_cntmνs = c_cgs ./ reverse(cntmλs) #frequencies at which to calculate the continuum
 
-    #sort the lines if necessary and check that λs is sorted
+    #sort the lines if necessary and 
     issorted(linelist; by=l->l.wl) || sort!(linelist, by=l->l.wl)
+    #discard lines far from the wavelength range being synthesized
+    linelist = filter(l-> λs[1] - line_buffer*1e-8 <= l.wl <= λs[end] + line_buffer*1e-8, linelist)
+    
+    #check that λs is sorted
     if step(λs) < 0
         throw(ArgumentError("λs must be in increasing order."))
     end
 
-    #discard lines far from the wavelength range being synthesized
-    linelist = filter(l-> λs[1] - line_buffer*1e-8 <= l.wl <= λs[end] + line_buffer*1e-8, linelist)
+    if A_X[1] != 12
```

Comment:
Since this is the main API function, I think we should be explicit about the behavior of synthesize, when `A_X` has the "wrong" size.

If it holds just 1 element, then we currently assume that the star purely consists of hydrogen... That's fine, but I feel like we need to coerce the length of `A_X` or more importantly of `abs_abundances` (so we avoid other problems later on). Alternatively, we could just raise an error of `A_X` has the wrong length.

Reply:
I've pushed a change to explicitly check that length(A_X) is right.  I only intended to support complete abundance specifications.